### PR TITLE
Deprecate Action ID-less backward compatibility

### PR DESCRIPTION
**Pull request by mendelB**
### What's this PR do?

This pull request deprecates the backward compatibility logic we'd set up across Photo & TextSubmissionActions and SocialShareActions to support blocks missing an Action ID for POSTing to Northstar.

### How should this be reviewed?
commit-by-commit
Did I miss any other places we've got logic set up?
Is the deprecation safe & sound?

### Any background context you want to provide?
Since [running the script](https://dosomething.slack.com/archives/C02BBP0CU/p1628790345001100) in #2698 and some manual backfilling (https://www.pivotaltracker.com/story/show/178059001/comments/226179030), we've removed any Contentful blocks from Production that would be missing an Action ID. We should be all set to finally retire this logic along with the API endpoint we were using for these blocks.

### Relevant tickets

References [Pivotal #178059001](https://www.pivotaltracker.com/story/show/178059001).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added screenshots of front-end changes on small, medium, and large screens.
- [x] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/actions/PhotoSubmissionAction/PhotoSubmissionAction.js [hunk 1]
```diff
@@ -240,18 +235,7 @@ class PhotoSubmissionAction extends PostForm {
...
-      this.props.storeCampaignPost(this.props.campaignId, {
-        action,
-        actionId: this.props.actionId,
-        blockId: this.props.id,
-        body: formatPostPayload(formFields),
-        pageId: this.props.pageId,
-        type,
-      });
-    }
+    this.props.storePost(data);
```
Comment (katiecrane) [inline resources/assets/components/actions/PhotoSubmissionAction/PhotoSubmissionAction.js:238]: this part is so exciting!!!

File: tests/Feature/StorePhotoReportbackPostTest.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ protected function setup(): void
     {
         parent::setUp();
 
-        $this->campaignId = 'XYZStringID123'; // @TODO temp
+        $this->actionId = '123'; // @TODO temp
```
Comment (katiecrane) [inline tests/Feature/StorePhotoReportbackPostTest.php:16]: Do we need this set? I don't think I see it used anywhere!

**Review by katiecrane — APPROVED** [review]
This looks INCREDIBLE!! Thank you Mendel for taking this on, it is a huge breath of fresh air to not deal with these two cases anymore.

File: tests/Feature/StorePhotoReportbackPostTest.php (hunk 2 above)
Comment (mendelB) [reply tests/Feature/StorePhotoReportbackPostTest.php]: Oops, truths! 🗑️ https://github.com/DoSomething/phoenix-next/pull/2700/commits/3295b9f9aa57bef561a7d9b7c89aac0bec7acf2d

Comment (katiecrane) [reply tests/Feature/StorePhotoReportbackPostTest.php]: I'm just here to delete more code 😈

**Pull request merged**